src/StoredEvents/Repositories/EloquentStoredEventRepository.php

```diff
@@ -129,13 +131,23 @@ public function update(StoredEvent $storedEvent): StoredEvent
 
     private function getEventClass(string $class): string
     {
-        $map = config('event-sourcing.event_class_map', []);
-
-        if (! empty($map) && in_array($class, $map)) {
-            return array_search($class, $map, true);
-        }
-
-        return $class;
+        $alias = collect(config('event-sourcing.event_class_map', []))
+            ->flip() // a given event should be aliased only once, so let's flip the collection to make class name the key
+            ->merge(
+                // merge alias from class' EventAlias attribute, if present
+                collect([new ReflectionClass($class)])
```

Comment:
Let's make this more readable, by retrieving and merging in the event alias in a separate function. By naming that new function right, you could lose this comment.

Reply:
Same remarks for the other changes, try to break it up in smaller functions (that will also make it easier for users to override stuff)

Reply:
Thanks Freek, I will revisit this as soon as possible.

Reply:
@freekmurze I've extracted a Dictionary facade since some of the logic was duplicated between StoredEvent and EloquentStoredEventRepository. Let me know if it makes sense.